src/vm/jitinterface.cpp

```diff
@@ -2313,10 +2357,9 @@ unsigned CEEInfo::getClassGClayout (CORINFO_CLASS_HANDLE clsHnd, BYTE* gcPtrs)
         }
         else
         {
-            // TODO-SPAN: Proper GC reporting
             memset(gcPtrs, TYPE_GC_NONE,
-                   (VMClsHnd.GetSize() + sizeof(void*) -1)/ sizeof(void*));
-            result = 0;
+                (VMClsHnd.GetSize() + sizeof(void*) - 1) / sizeof(void*));
+            result = ComputeGCLayout(VMClsHnd.AsMethodTable(), gcPtrs);
```

Comment:
Is the GC layout computation here more complex than the typed reference case because these new types can be embedded in other value classes and typed references cannot? If so it would be worth a comment at least.